python/ray/train/torch/config.py

```diff
@@ -95,12 +95,17 @@ def _setup_torch_process_group(
         )
     logger.debug(f"using {backend}")
 
-    if backend == "nccl" and "NCCL_BLOCKING_WAIT" not in os.environ:
+    # See the `timeout` arg in https://pytorch.org/docs/master/
+    # distributed.html#torch.distributed.init_process_group for description of
+    # NCCL_ASYNC_ERROR_HANDLING. We do not use NCCL_BLOCKING_WAIT due to performance
+    # overhead.
+    if backend == "nccl" and "NCCL_ASYNC_ERROR_HANDLING" not in os.environ:
```

Comment:
Seems like we should respect `Only one of these two environment variables should be set.` from the docs you link, e.g. not set this if `NCCL_BLOCKING_WAIT` is set.

Reply:
good point! updated